src/python/pants/core/subsystems/debug_adapter.py

```diff
@@ -23,3 +25,13 @@ class DebugAdapterSubsystem(Subsystem):
         default=5678,
         help="The port to use when launching the server.",
     )
+
+    @staticmethod
+    def port_for_testing() -> int:
```

Comment:
I'm not a fan of polluting normal code for testing sake.

Why do it this way and not using the car directly? Or better yet, in a high level contest you can monkeypatch the debugadapter default. Then tests don't need modifications and all new tests are also safe

Reply:
What is the car?

Reply:
I don't know where and how that monkeypatching would happen? I'm happy to turn this over to you if you have a clear idea, I just want these tests to not be broken...

Reply:
Sorry, typo.

And yeah I can, but I'm not in front of a tower until tuesday

Reply:
Ideally you could use port 0 but it looks like they did not set debugpy up for this. I personally demand that of all my port-using tests in any project in any language, but it may require some extra effort here - possibly too brittle. FWICT you'd have to info-log scrape / poll :/.

Reply:
Yeah, alas, that was my first choice too, and I agree.

Reply:
@thejcannon how about I merge this, and we (you?) can improve it in the future if the mood strikes. Right now this fixes a bug in our tests, so it is strictly better than the status quo ante.

Reply:
In that case I'd rather we just inline the port selection code so we still aren't dirtying our "prod" code

Reply:
I don't want to inline it directly, but I moved the function to a util.